labonneboite/tests/web/api/test_api.py

```diff
@@ -295,6 +295,13 @@ def test_query_returns_scores_adjusted_to_rome_code_context(self):
                 'rome_codes': rome_code,
                 'user': u'labonneboite',
             })
+
+            # FIXME mock.patch
+            # The example in this test has a very low score under the usual SCORE_FOR_ROME_MINIMUM,
+            # which is why we need to lower the threshold just for this test, in order
+            # to be able to compute back the score from the stars.
+            scoring_util.SCORE_FOR_ROME_MINIMUM = 0.0
```

Comment:
with @mock.patch('labonneboite.common.scoring', 'SCORE_FOR_ROME_MINIMUM', 0):
       ...

ici il faut utiliser `mock.patch` et non `mock.patch.object`.

Reply:
mmm pas vraiment compris et ne fonctionne pas. (le @ est une typo, correct?)

J'ai aussi tenté des trucs du genre:

`with mock.patch.object(self.app.Office.scoring_util, 'SCORE_FOR_ROME_MINIMUM', 0):`

pour pouvoir mocker `self.app` dans `self.app.get('%s?%s' % (url_for("api.company_list"), urlencode(params)))` qui doit utiliser ma valeur hackée de SCORE_FOR_ROME_MINIMUM

mais en fait l'objet `self.app` n'a pas le champ `Office`, c'est un objet FlaskClient et non pas juste le module app.py, logique. J'ai creusé mais je ne sais pas comment rentrer dans cet object FlaskClient et lui dire d'aller mocker son Office.scoring_util

en attendant je vais faire un try eventually moche.... :-/

ce serait cool si on arrive à fixer ça avant que tu partes @regisb

Reply:
cf https://github.com/StartupsPoleEmploi/labonneboite/commit/4b89251656302270246d96d1d1198120c050bc4e @regisb

Reply:
j'essaie encore un truc

Reply:
kudos @regisb the mock master

Reply:
done